gateway-js/src/index.ts

```diff
@@ -863,4 +865,5 @@ export {
   Experimental_CompositionInfo,
 };
 
+export * from './QueryPlan';
```

Comment:
In the past, this `export *` pattern has led to over-exporting and over-exposing internal details which often end up being used by consumers of this package.  Apollo Server is a really strong example of how that can end up! Auto-complete makes it very easy for users to find these things and it tends to make the auto-complete more noisy rather than just showing the most user-applicable bits (i.e. when importing the package) which also makes it more likely that someone might import something that's an implementation detail inadvertently.

If the motivation for `export *` is entirely for testing, might we consider a pattern that avoids exporting all of `QueryPlan`, similar to what I did in `apollo-cache-control` in https://github.com/apollographql/apollo-server/commit/68cbc938ff709078efc8a1e726300e61222f6b8b#diff-c209a58137b4a9da2beba783d377bff6c9c057208251f4dbea56763751680a80R442-R445 of https://github.com/apollographql/apollo-server/pull/3997 where I used a `__testing__` object to expose things that were _specifically_ for testing?  This groups all the testing stuff together in auto-complete, and also provides a more clear understanding to the user that this is not a public API.

**If this is _not_ for testing**, but instead intended to surface public API, could we be explicit about what we're intending on exporting so that newly-`export`ed properties of the `./QueryPlan` package don't get inadvertently exported and polluting the root module?